doc/api/packages.md

```diff
@@ -434,16 +499,21 @@ All instances of `*` on the right hand side will then be replaced with this
 value, including if it contains any `/` separators.
 
 ```js
-import featureX from 'es-module-package/features/x';
+import featureX from 'es-module-package/features/x.js';
 // Loads ./node_modules/es-module-package/src/features/x.js
 
-import featureY from 'es-module-package/features/y/y';
+import featureY from 'es-module-package/features/y/y.js';
 // Loads ./node_modules/es-module-package/src/features/y/y.js
 
-import internalZ from '#internal/z';
+import internalZ from '#internal/z.js';
 // Loads ./node_modules/es-module-package/src/internal/z.js
 ```
 
+Including the `"*.js"` on both sides of the mapping allows restricting which
+file extensions can be resolved in the case of other files like source maps
+existing, while also retaining file extensions for the consumed subpath as is
+recommended.
```

Comment:
i don't think node should be making this recommendation at all.

Reply:
I've added a new dedicated section on interoperability with import maps and linked to that, this makes it an interoperability note and choice, based on being informed about the interoperability concern. Instead of saying "extensions are recommended" instead the guidance then becomes "if interoperability with import maps is desired, extensions are recommended".